src/diffusers/configuration_utils.py

```diff
@@ -162,6 +170,21 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         self.to_json_file(output_config_file)
         logger.info(f"Configuration saved in {output_config_file}")
 
+        # Additionally, save the implementation file too. It can happen for a pipeline, for a model, and
+        # for a scheduler.
+        if self.__class__.__name__ not in _all_available_pipelines_schedulers_model_classes:
+            module_to_save = self.__class__.__module__
+            absolute_module_path = sys.modules[module_to_save].__file__
+            try:
+                with open(absolute_module_path, "r") as original_file:
+                    content = original_file.read()
+                path_to_write = os.path.join(save_directory, f"{module_to_save}.py")
+                with open(path_to_write, "w") as new_file:
+                    new_file.write(content)
+                    logger.info(f"{module_to_save}.py saved in {save_directory}")
+            except Exception as e:
+                logger.error(e)
```

Comment:
Not sure why @patrickvonplaten's comment got resolved, but Patrick has a concern about the implications of serializing modules like this in forked `diffusers` repos. 

To that, I responded with the following. 

If we don't expose a way to let the users serialize these files it might become cumbersome to do certain changes manually to make `trust_remote_code` foolproof. The issue becomes more evident for pipelines having more components (SDXL, is a good example). The developer experience becomes a bit untidy here IMO. 

To make this safer, we can maybe expose an argument called `save_custom_modules` in `save_pretrained()` and default that to False. This way, users are fully aware of what they're doing. 

But apart from that, I think this path won't be hit as often as one might think unless I am missing something.

Reply:
@sayakpaul If it were me, I would definitely use custom pipeline class to call save_pretrained, so the save_custom_modules parameter might be a bit redundant.